### ISSUE TITLE
Updated to 15.9p5, now VS prompts to upgrade many projects

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#31053]
Just took the latest update offered to me.  when i try to open roslyn.sln i get: 

![image](https://user-images.githubusercontent.com/4564579/48234543-dc7eb800-e36e-11e8-8966-179427a33559.png)

I have don't a git-clean, and a restore already.

**Comment by CyrusNajmabadi** [comment]
Tagging @jaredpar @sharwell @jasonmalinowski

**Comment by Pilchie** [comment]
Can you run `devenv /updateconfiguration` once?

**Comment by jasonmalinowski** [comment]
And it might not also hurt to let it do that "one way" conversion, commit the diff somewhere and send it over. Not that we'll take it, but might help to investigate.

**Comment by CyrusNajmabadi** [comment]
Interesting.  i did that, now it opens.  but none of the VB projects loaded...

wtf :)

![image](https://user-images.githubusercontent.com/4564579/48236937-47cd8780-e379-11e8-8eee-4cc9fd501d61.png)

**Comment by CyrusNajmabadi** [comment]
Oh, if i try to reload one of those projects, i get:

![image](https://user-images.githubusercontent.com/4564579/48236958-5ddb4800-e379-11e8-8521-68b1e080cd4f.png)

Thoughts?

**Comment by CyrusNajmabadi** [comment]
I'm also getting this: 

```
C:\github\roslyn\build\Targets\RepoToolset\Version.targets(44,5): error MSB4044: The "Microsoft.DotNet.Arcade.Sdk.CheckRequiredDotNetVersion" task was not given a value for the required parameter "SdkVersion". [C:\github\roslyn\src\Deployment\RoslynDeployment.csproj]
```

**Comment by CyrusNajmabadi** [comment]
Note: the 'cannot evaluate' error is te same i got here: https://github.com/dotnet/roslyn/issues/30863

**Comment by CyrusNajmabadi** [comment]
Same issue:

```
C:\github\roslyn [master ≡]> .\Build.cmd
Repo Dir C:\github\roslyn
Binaries Dir C:\github\roslyn\Binaries
C:\github\roslyn\build\Targets\RepoToolset\Version.targets(44,5): error MSB4044: The "Microsoft.DotNet.Arcade.Sdk.Check
RequiredDotNetVersion" task was not given a value for the required parameter "SdkVersion". [C:\github\roslyn\src\Deploy
ment\RoslynDeployment.csproj]
C:\github\roslyn\build\Targets\RepoToolset\Version.targets(44,5): error MSB4044: The "Microsoft.DotNet.Arcade.Sdk.Check
RequiredDotNetVersion" task was not given a value for the required parameter "SdkVersion". [C:\github\roslyn\src\Compil
ers\CSharp\Test\Syntax\Microsoft.CodeAnalysis.CSharp.Syntax.UnitTests.csproj]
```

**Comment by CyrusNajmabadi** [comment]
Sam helped me out hre . I needed to install: https://www.microsoft.com/net/download/dotnet-core/2.2

As before, it would be *great* if Roslyn could catch this and tell the user what to do.

**Comment by jaredpar** [comment]
Agree on the frustrating bad error here. But it’s not something Roslyn can, or should solve. It’s **extremely** frustrating to me too.

**Comment by CyrusNajmabadi** [comment]
Ok... but how would someone normally figure this out without a Sam or Jason or you to talk to? :)   You update VS... and Roslyn stops building with *extremely* cryptic error messages.  It seems like Roslyn could maybe do some sort of check initially for these sorts of things and at least give a warning about something needed that you might need to install.

**Comment by jaredpar** [comment]
I have no idea how others would do it.

Yes Roslyn could add a check. I’ve writtwn such a check several times. But that does noting to further the .NET experience. It is just locally patching a problem. Every other repo would still have this issue.

**Comment by jaredpar** [comment]
I’m following up internally about this.

**Comment by CyrusNajmabadi** [comment]
thanks!

**Comment by alrz** [comment]
> I needed to install: https://www.microsoft.com/net/download/dotnet-core/2.2

Shouldn't we update `Building, Debugging, and Testing on Windows.md` when the sdk changes? It seems like it's out of date on master.

**Comment by jaredpar** [comment]
> Shouldn't we update Building, Debugging, and Testing on Windows.md when the sdk changes? It seems like it's out of date on master.

Yes they should absolutely be kept in sync. The thing is though they should be in sync: i'm generally the one who changes the SDK dependency and I also change that document too when it happens. I'm not aware of our dependency changing here anytime recently though. 

Did it change and I just missed it? I was out for half a week.

**Comment by jaredpar** [comment]
@CyrusNajmabadi if you feel passionate about this I suggest you leave your feedback on the dotnet/sdk repository. I've given as much feedback as I can to the team about the pain and frustration that this behavior causes. Not just in recent times, but over essentially the last year. I'm not sure my feedback is going to move the needle at this point.

**Comment by CyrusNajmabadi** [comment]
Is there an issue you can link to?

**Comment by alrz** [comment]
>Did it change and I just missed it? I was out for half a week.


I don't know. I just saw Cyrus's [comment](https://github.com/dotnet/roslyn/issues/31053#issuecomment-437232421). Nearly every time I try to rebase to master and we have a new sdk I see some errors that don't tell me what's wrong. I put a couple of hours aside whenever I'm going to merge/rebase to master when I know sdk has been updated. I'd usually start by reading the .md file I mentioned, if that hasn't been updated I'd end up in a confused state and need to try everything.

**Comment by jaredpar** [comment]
> Is there an issue you can link to?

I'm sure there is one somewhere. I looked and couldn't find it. Possibly @nguerrera, @dsplaisted know what it is.

**Comment by CyrusNajmabadi** [comment]
@nguerrera @dsplaisted Can you give guidance on the best way forward here?  Numerous times after updating VS I (and many other devs) are left in a non-working state where many parts of the build and project systems just spit out incredibly cryptic and non-actionable messages (see this bug and https://github.com/dotnet/roslyn/issues/30863 for examples).  

In all these cases, it's necessary for some team member to come along and inform people on what to do (usually uninstalling/reinstalling some dotnet package).  This is a pretty bad state of affairs, and is veyr unfriendly to the community.

@jaredpar Says he has asked for efforts to be made in .net to improve things here.  I'm not sure what the specific ask was for, but i would hope at a minimum it would be making it so that the user gets actionable error messages.  

Could you help out here and get this issue in front of the right people so this problem can be resolved?

Thanks!

**Comment by CyrusNajmabadi** [comment]
> We can update the task in repo toolset to detect this

That wouldbe great.  Would it be possible for the detection to give a message about what the root cause/fix would be?  Thanks!

**Comment by nguerrera** [comment]
Yes, that's possible. I will tag you on the change to review the message.

**Comment by nguerrera** [comment]
> I'm unsure if they're not working or just not in this branch yet. 

They're not working, this is coming from them:

```
C:\github\roslyn\build\Targets\RepoToolset\Version.targets(44,5): error MSB4044: The "Microsoft.DotNet.Arcade.Sdk.Check
RequiredDotNetVersion" task was not given a value for the required parameter "SdkVersion". [C:\github\roslyn\src\Deploy
ment\RoslynDeployment.csproj]
```

> Given the product team doesn't see this as worth fixing in dev15

Do you have a way to get risky changes into dev*15* that I don't know about?

**Comment by CyrusNajmabadi** [comment]
>@CyrusNajmabadi Which branch of Roslyn are you building?

"master"

> But then 9 days ago , it was identified that you had one of the bad 2.2.0-* builds and you presumably uninstalled it then

Yup.

> so I don't understand how you then hit this again today. 

@jaredpar and i tend to wake up in the morning, and then everything goes downhill.  

>  You say that you upgraded to 15.9p5, but that would have given you a 2.1.500-preview build, not a bad 2.2.0-* build. In the intervening time, did you by any chance build a different branch that still thinks you need exactly 2.2.0-*, and would have instructed you to re-install it?

I don't think i ever manually did anything.  However, i'm *commonly* jumping around a ton of branches in roslyn running "restore" (which i think does a "dotnet restore").  So maybe that could have done it?

**Comment by CyrusNajmabadi** [comment]
@nguerrera I hit this again with 15.9p6.  Same error:

![image](https://user-images.githubusercontent.com/4564579/48880157-f59e5480-edc3-11e8-97bc-3166cf0e9fa7.png)

Is tehre something about VS updates that break this every time?

**Comment by nguerrera** [comment]
In your VS command prompt, can you run `dotnet --info` and send me the output?

**Comment by nguerrera** [comment]
(To answer the question, nothing known should be causing this on every VS update, but clearly something unknown is happening to you.)

**Comment by nguerrera** [comment]
Are you still also getting the CheckRequiredDotnetVersion error?

**Comment by nguerrera** [comment]
> `C:\Program Files (x86)\Microsoft Visual Studio\2017\Community>`

That looks like VS 15.9 *RTM* judging by the default installation directory, which now disables preview SDKs by default. 

See https://blogs.msdn.microsoft.com/dotnet/2018/11/13/net-core-tooling-update-for-visual-studio-2017-version-15-9/

If that's right, you can use the preview SDK you have installed by opting in to previews via Tools -> Options -> Projects and Solution -> .NET Core -> Use previews of the .NET Core SDK

However, the fact that you have *only* 2.2.100-preview3 installed suggests that you do not have the VS .NET Core workload installed because it would have installed a 2.1.500 RTM SDK that would have avoided these errors. 

I suggest opening the VS installer, and selecting the .NET Core workload. You can then uninstall the 2.2.100 SDK leaving only the 2.1.500 SDK that came with VS in add/remove programs. That should get you back to a vanilla state moving forward.

We are making improvements to this in VS 16 that should prevent this state:

1. .NET Core SDK in all .NET workloads (not just .NET Core and ASP.NET)
2. Removing the fallback v1.1 SDK that existed for the other workloads to use .NET Standard, but caused errors that should indicate that the SDK could not be located to surface as old v1.1 bugs instead. In this case the VB projects are failing to evaluate due a bug in how VB documentation files were set in SDK v1.1.

I will also follow up as promised to make the custom arcade version check more robust while we wait for better handling in the product in VS 16.

**Comment by alrz** [comment]
>We are making improvements to this in VS 16 that should prevent this state:

Will this be available in preview? (1 or 2?) Is there an issue that we could follow the progress? thanks

**Comment by nguerrera** [comment]
dotnet/cli#6795 (linked above as well as related project system issue).

**Comment by CyrusNajmabadi** [comment]
> I suggest opening the VS installer, and selecting the .NET Core workload. 

I have that workload installed.  For both teh normal VS install and the VS-preview install.

**Comment by nguerrera** [comment]
It's unexpected that you wouldn't have 2.1.500 then.  Something is going wrong in VS setup.

Can you run https://www.microsoft.com/en-us/download/details.aspx?id=12493 

Email me the results. nicholg @ microsoft. I won't be able to investigate until Monday.

**Comment by nguerrera** [comment]
After collecting logs, I would try repair in VS installer.
